CONTRIBUTING.md

```diff
@@ -3,196 +3,112 @@
+1. Install [conda-develop](https://docs.conda.io/projects/conda-build/en/latest/resources/commands/conda-develop.html):
+    ```
+    conda install --name base conda-develop
+    ```
 
-Contributing to ilastik is a little bit more involved than just cloning the repo and committing.
-The two main repositories; ilastik, and volumina are governed by ilastik-meta as submodules.
-Furthermore, in order to be able to run the code, all dependencies have to be installed.
-By far the most convenient way for this is to use conda.
-The following paragraphs will guide through the process of setting up a working development environment.
+1. Create a new environment and install dependencies:
+    ```
+    conda deactivate
+    conda env remove --name ilastik
+    conda create --name ilastik --channel ilastik-forge --channel conda-forge ilastik-dependencies-no-solvers pre-commit
+    ```
 
-### Fork the relevant repositories
+1. Install repositories as packages in development mode:
+    ```
+    conda develop --name ilastik volumina
+    conda develop --name ilastik ilastik
+    ```
 
-You should begin by forking the two main ilastik repositories, that make up the ilastik application:
+1. Install pre-commit hooks:
+    ```
+    conda activate ilastik
 
-* https://github.com/ilastik/ilastik
-* https://github.com/ilastik/volumina
+    cd volumina
+    pre-commit install
+    cd ..
 
-#### Clone ilastik-meta
+    cd ilastik
+    pre-commit install
+    cd ..
+    ```
 
-Having done that, you should clone our `ilastik-meta` repository with:
+1. Launch ilastik:
+    ```
+    conda activate ilastik
+    cd ilastik
+    python ilastik.py
+    ```
 
-```bash
-# navigate to an appropriate folder, e.g. `~/sources`
-git clone https://github.com/ilastik/ilastik-meta
-```
+## Workflow
 
-#### Initialize the source tree with your own forks
+We use [GitHub Flow](https://guides.github.com/introduction/flow/) workflow without the _Deploy_ step.
 
-Examining the created folder, e.g. `~/sources/ilastik-meta`, you can find the `.gitmodules` file.
-Check out a new branch in ilastik-meta, e.g. `git checkout -b my-forks` and edit the `.gitmodules`
-file such that it points to your own forks: replace all occurrences of `https://github.com/ilastik`
-with `https://github.com/<your_github_username>`.
+1. Sync your local and remote _master_ branches with the upstream.
+    ```
+    git -C volumina pull --ff-only upstream master:master
+    git -C volumina push origin master:master
 
-Now it is time to initialize the repository:
+    git -C ilastik pull --ff-only upstream master:master
+    git -C ilastik push origin master:master
+    ```
 
-```bash
-# in ~/sources/ilastik-meta (or wherever you have cloned ilastik-meta to)
-git submodule update --init --recursive
-git submodule foreach 'git checkout master'
-```
+1. Switch to new branches:
+    ```
+    git -C volumina checkout -b your-branch-name-here master
```

Comment:
this seems to imply that you _always_ checkout a branch on both, which is seldom the case.